transformer_lens/HookedTransformer.py

```diff
@@ -1470,20 +1471,39 @@ def generate(
 
         assert isinstance(tokens, torch.Tensor)
         batch_size, ctx_length = tokens.shape
-        tokens = tokens.to(devices.get_device_for_block_index(0, self.cfg))
+        device = devices.get_device_for_block_index(0, self.cfg)
+        tokens = tokens.to(device)
         if use_past_kv_cache:
             past_kv_cache = HookedTransformerKeyValueCache.init_cache(
                 self.cfg, self.cfg.device, batch_size
             )
         else:
             past_kv_cache = None
 
-        if stop_at_eos and eos_token_id is None:
-            assert (
+        stop_tokens = []
+        pad_token_id = 0
```

Comment:
Also, can well call this `eos_token_for_padding`? HuggingFace uses `pad_token_id` and we don't so reading this code was pretty confusing